File: bod.php

<?php
// koneksi database
$conn = new mysqli("localhost", "root", "", "db_bwm");

// ambil id yang dikirim lewat URL
$id = $_GET['id'];

// ambil data berdasarkan id
$result = $conn->query("SELECT * FROM table_client WHERE Id = $id");
$row = $result->fetch_assoc();
?>

<!DOCTYPE html>
<html>
<head>
  <title>BOD Capacity</title>
</head>
<body>
<h2>BOD Capacity</h2>
<form method="post" action="update.php">
  <!-- kirim id sebagai hidden -->
  <input type="hidden" name="id" value="<?php echo $row['Id']; ?>">

  Hostname:<br>
  <input type="text" name="hostname" value="<?php echo $row['Hostname']; ?>" readonly><br><br>

  Description:<br>
  <input type="text" name="description" value="<?php echo $row['Description']; ?>" readonly><br><br>
  
  Interface:<br>
  <input type="text" name="interface" value="<?php echo $row['Interface']; ?>" readonly><br><br>
    
  Unit:<br>
  <input type="text" name="unit" value="<?php echo $row['Unit']; ?>" readonly><br><br>

  Input Policer Existing:<br>
  <input type="text" name="input_policer" value="<?php echo $row['Input_policer']; ?>" readonly><br><br>

  Output Policer Existing:<br>
  <input type="text" name="output_policer" value="<?php echo $row['Output_policer']; ?>" readonly><br><br>

  Input Policer BOD:<br>
  <input type="text" name="bod_input_policer" value=""><br><br>

  Output Policer BOD:<br>
  <input type="text" name="bod_output_policer" value=""><br><br>
  
  BOD Until:<br>
  <input type="datetime-local" name="bod_time" value=""><br><br>

  <input type="submit" value="Update">
</form>
</body>
</html>

File: bod.py
#!/usr/bin/env python3
import sys
from jnpr.junos import Device
from jnpr.junos.utils.config import Config
import mysql.connector

# ambil input dari user
device_ip = sys.argv[1]
username = "juniper"
password = "juniper123"
iface_name = sys.argv[2]
unit_name = sys.argv[3]
policer_input_value = sys.argv[4]
policer_output_value = sys.argv[5]
hostname = sys.argv[6]
description = sys.argv[7]
old_input_policer = sys.argv[8]
old_output_policer = sys.argv[9]
datetime = sys.argv[10]

# buka koneksi ke device
with Device(host=device_ip, user=username, passwd=password) as dev:
    with Config(dev, mode='exclusive') as cu:
        # Buat konfigurasi dalam format XML
        config_xml = f"""
        <configuration>
          <interfaces>
            <interface>
              <name>{iface_name}</name>
              <unit>
                <name>{unit_name}</name>
                <family>
                  <inet>
                    <policer>
                      <input>{policer_input_value}</input>
                      <output>{policer_output_value}</output>
                    </policer>
                  </inet>
                </family>
              </unit>
            </interface>
          </interfaces>
        </configuration>
        """
        # load configuration
        cu.load(config_xml, format='xml')
        # commit perubahan
        cu.commit()
        print(f"Policer input/output berhasil diset pada {iface_name} unit {unit_name}.")
try:
    conn = mysql.connector.connect(
        host='localhost',
        user='root',
        password='',
        database='db_bwm'
    )
    cursor = conn.cursor()

    sql = """INSERT INTO table_bod 
             (Hostname, Description, Interface, Unit, Old_input_policer, Old_output_policer, Bod_input_policer, Bod_output_policer, datetime)
             VALUES (%s,%s,%s,%s,%s,%s,%s,%s,%s)"""

    data_insert = (
        hostname,
        description,
        iface_name,
        unit_name,
        old_input_policer,
        old_output_policer,
        policer_input_value,
        policer_output_value,
        datetime,
    )

    cursor.execute(sql, data_insert)

    sql_update = """
    UPDATE table_client
    SET Input_policer=%s,
        Output_policer=%s
    WHERE Hostname=%s AND Interface=%s AND Unit=%s
    """
    data_update = (policer_input_value, policer_output_value, hostname, iface_name, unit_name)
    cursor.execute(sql_update, data_update)

    conn.commit()
    print("Data perubahan policer berhasil disimpan ke database table_bod.")
except Exception as e:
    print(f"Gagal menyimpan ke database: {e}")
finally:
    if cursor:
        cursor.close()
    if conn:
        conn.close()